tests/cli/wit-directory-output-in-deps-folder-with-multiple-version.wit

```diff
@@ -1,16 +0,0 @@
-// RUN: component wit % --out-dir %tmpdir | component wit %tmpdir
```

Comment:
Could this test be preserved with a new package header at the top?

Reply:
Yeah I've been trying to decide what to do about this test, as well as what the desired behavior even is for this command.

Right now `DecodedWasm` doesn't differentiate between whether packages are from a deps folder or nested.  We could add this differentiation.
 
As I see it at the moment, there are a few options available:

1.) Capture if a package is nested or a deps folder package when we resolve, so we know how the package should be printed.  I think this would also require storing this info in binaries?  Which feels like something we may not want to do?

2.) Every package that isn't the root is nested... no deps folder is printed.

3.) Every package that isn't the root is printed in a deps folder... no nested packages (though packages in the original deps folder can have nested package definitions, so those would be nested still)

4.) Flags determine behavior

Maybe I'm missing an obvious alternative...

Reply:
Ah good points, but I think it's ok to create a `deps` folder when there wasn't one before. Organization of WIT might vary but the tooling here splatting out a single way to represent things I think makes sense. This is mostly just a test that whatever `--out-dir` produces is readable by `component wit`.

Implementation-wise I think it's fine to either shove everything into a `deps/*.wit` file or even just use `package { ... }` for all deps in the root folder. The whole `--out-dir` argument dosn't make a ton of sense any more nowadays that there's multi-package-forms-in-one-file but I figure  it's still good to not break it in the meantime.